aea/cli/generate.py

```diff
@@ -45,69 +55,118 @@ def generate(
 
 
 @clean_after
-def _generate_item(ctx: Context, item_type: str, specification_path: str) -> None:
-    """Generate an item based on a specification and add it to the configuration file and agent."""
-    # Get existing items
-    existing_id_list = getattr(ctx.agent_config, "{}s".format(item_type))
-    existing_item_list = [public_id.name for public_id in existing_id_list]
-
-    item_type_plural = item_type + "s"
+def _generate_protocol(ctx: Context, protocol_specification_path: str) -> None:
+    """Generate a protocol based on a specification and add it to the configuration file and agent."""
+    protocol_plural = PROTOCOL + "s"
 
-    # Load item specification yaml file
+    # Create protocol generator (load, validate,
+    # extract fields from protocol specification yaml file)
     try:
-        protocol_spec = load_protocol_specification(specification_path)
-    except Exception as e:
-        raise click.ClickException(str(e))
-
-    protocol_directory_path = os.path.join(
-        ctx.cwd, item_type_plural, protocol_spec.name
-    )
+        output_path = os.path.join(ctx.cwd, protocol_plural)
+        protocol_generator = ProtocolGenerator(protocol_specification_path, output_path)
+    except FileNotFoundError as e:
+        raise click.ClickException(  # pragma: no cover
+            "Protocol is NOT generated. The following error happened while generating the protocol:\n"
+            + str(e)
+        )
+    except yaml.YAMLError as e:
+        raise click.ClickException(  # pragma: no cover
+            "Protocol is NOT generated. The following error happened while generating the protocol:\n"
+            + "Yaml error in the protocol specification file:"
+            + str(e)
+        )
+    except ProtocolSpecificationParseError as e:
+        raise click.ClickException(  # pragma: no cover
+            "Protocol is NOT generated. The following error happened while generating the protocol:\n"
+            + "Error while parsing the protocol specification: "
+            + str(e)
+        )
+    except Exception as e:  # pragma: no cover
+        raise click.ClickException(  # pragma: no cover
+            "Protocol is NOT generated. The following error happened while generating the protocol:\n"
+            + str(e)
+        )
 
-    # Check if we already have an item with the same name in the agent config
+    # helpers
+    language = ctx.config.get("language")
+    existing_protocol_ids_list = getattr(ctx.agent_config, "{}s".format(PROTOCOL))
+    existing_protocol_name_list = [
+        public_id.name for public_id in existing_protocol_ids_list
+    ]
+    protocol_spec = protocol_generator.protocol_specification
+    protocol_directory_path = os.path.join(ctx.cwd, protocol_plural, protocol_spec.name)
     logger.debug(
         "{} already supported by the agent: {}".format(
-            item_type_plural, existing_item_list
+            protocol_plural, existing_protocol_name_list
         )
     )
```

Comment:
Remove

Reply:
The whole `logger.debug` block?

Reply:
Yes, it's factually incorrect in this position.